src/sidebar/config/build-settings.js

```diff
@@ -178,55 +130,78 @@ async function fetchGroupsAsync(config, rpcCall) {
- * @param {Window} window_ - Test seam.
- * @return {Promise<MergedConfig>} - The merged settings.
+ * In cases where configuration is retrieved by RPC, the `configFromURL` object
+ * values are not retained in the returned host configuration. Note the
+ * exception related to the `group` property below.
+ *
+ * @param {HostConfig} configFromURL
+ * @param {Window} [window_]
+ * @returns {Promise<HostConfig>}
  */
-export async function fetchConfig(appConfig, window_ = window) {
-  const hostConfig = hostPageConfig(window);
-
-  const requestConfigFromFrame = hostConfig.requestConfigFromFrame;
+async function findHostConfiguration(configFromURL, window_) {
+  const requestConfigFromFrame = configFromURL.requestConfigFromFrame;
 
   if (!requestConfigFromFrame) {
-    // Directly embed: just get the config.
-    return fetchConfigEmbed(appConfig, hostConfig);
+    // In this case, the `configFromURL` is retained and used as host configuration
+    return configFromURL;
   }
+
   if (typeof requestConfigFromFrame === 'string') {
-    // Legacy: send RPC to all parents to find config. (deprecated)
-    // nb. Browsers may display errors in the console when messages are sent to frames
-    // that don't match the origin filter".
-    return await fetchConfigLegacy(appConfig, window_);
+    return await fetchConfigFromAncestorFrame(requestConfigFromFrame, window_);
   } else if (
     typeof requestConfigFromFrame.ancestorLevel === 'number' &&
     typeof requestConfigFromFrame.origin === 'string'
   ) {
-    // Know parent frame: send RPC directly to the parent.
-    const parentFrame = getAncestorFrame(
-      requestConfigFromFrame.ancestorLevel,
-      window_
-    );
-
-    const rpcMergedConfig = await fetchConfigRpc(
-      appConfig,
-      parentFrame,
+    const hostConfiguration = await fetchHostConfigRPC(
+      getAncestorFrame(requestConfigFromFrame.ancestorLevel, window_),
       requestConfigFromFrame.origin
     );
-    // Add back the optional focused group id from the host page config
-    // as this is needed in the Notebook.
-    return {
-      ...rpcMergedConfig,
-      ...(hostConfig.group ? { group: hostConfig.group } : {}),
-    };
+
+    // Manually add a `group` property to the host configuration if
+    // it was present on `configFromURL`. This allows a focused group to be
+    // available to the Notebook within LMS contexts.
+    // FIXME: This could be organized/typed more elegantly.
+    if (configFromURL.group && !hostConfiguration.group) {
+      hostConfiguration.group = configFromURL.group;
+    }
+    return hostConfiguration;
   } else {
     throw new Error(
       'Improper `requestConfigFromFrame` object. Both `ancestorLevel` and `origin` need to be specified'
     );
   }
 }
+
+/**
+ * Build a settings object for use in the sidebar by extending provided base
+ * configuration with the correct "host configuration" settings.
+ *
+ * @param {SidebarConfig} sidebarConfiguration
```

Comment:
I wonder if we can find alternative names for `SidebarConfig` and `HostConfig` that better communicate what they mean. The prefix denotes where the configuration comes from rather than who it is intended for, which may not be obvious. Maybe `ConfigFromSidebar` / `ConfigFromHost`?

Reply:
I was thinking the same! Didn't want to take giant leaps all at once here but yeah, `HostConfig` is...not helpful as a name.